actions/rules/action_state_to_metastate.go

```diff
@@ -234,21 +238,35 @@ func (act ActionStateToMetaState) addOrUpdateChange(changes []convert.Change, at
 	return newChanges
 }
 
+func (act ActionStateToMetaState) fuseChanges(c1 []convert.Change, c2 []convert.Change) *[]convert.Change {
+	for _, change := range c2 {
+		c1 = act.addOrUpdateChange(c1, change.AttributeName, change.OldValue, change.NewValue)
+	}
+	return &c1
+}
+
 // OnChange executes the action rule.
 func (act ActionStateToMetaState) OnChange(newContext convert.ChangeDetector, contextChanges []convert.Change, configuration string, actionChanges *[]convert.Change) (convert.ChangeDetector, []convert.Change, error) {
 	if len(contextChanges) == 0 {
 		// no changes, just return what we have.
 		return newContext, *actionChanges, nil
```

Comment:
You're not checking for `actionChanges` being `nil` and instead just dereference it.

Reply:
If this is nil, something fundamentally went wrong. But anyway, fixed.